WordPress/gutenberg
# Block Bindings: Allow editing in post meta source

## What?
Use the latest changes to the block bindings API in the editor to allow users to edit the value of the custom fields directly through the blocks when they are connected. For example, when the paragraph content is bound to a specific custom field, and the user start typing, they would be modifying the custom field and not the content.

At least in this first iteration, it should be limited to:
* Users with the permissions to edit the field.
* When users are editing a post, and not a template.
* Custom fields should not be protected and available through the REST API.

## How?
Using the `setValue` function based on the code created by @ellatrix  [here](https://github.com/WordPress/gutenberg/pull/60725). Apart from that, we should use the `lockAttributesEditing` callback added in this pull request to cover the mentioned limitation.

Additionally, it ensure that the split selection work as expected in bound blocks.

## Testing Instructions
I added a couple of e2e tests for this use case. You can test it manually by:

## Testing Instructions




<details>
<summary>1. Register post meta by adding this snippet to your theme's functions.php</summary>

```php
add_action( 'init', 'test_block_bindings' );

function test_block_bindings() {
	register_meta(
		'post',
		'text_field',
		array(
			'show_in_rest'      => true,
			'single'            => true,
			'type'              => 'string',
			'default'           => 'default text value',
		)
	);
}
```

</details>


<details>

<summary>2. Add a paragraph block bound to the custom field using the Code Editor</summary>

```

<p>Paragraph content</p>

```

</details>
 
3. Check that the value of the custom field is shown.
4. Check you can edit it when you are in the post editor.
5. Check that modifying the value actually changes the custom field and not the paragraph content.
